tools/commands.py

```diff
@@ -85,19 +86,76 @@ def __init__(self, cmd_str):
+            target_args = []
+            other_filter_args = []
+            on_found = False
+            for arg in cmd_as_list[1:]:
+                if arg.startswith('on:'):
+                    on_found = True
+                    # Extract everything after 'on:' and split by comma
+                    filter_content = arg[3:]  # Remove 'on:' prefix
+                    target_args.extend(filter_content.split(','))
+                elif arg.startswith('for:'):
+                    # Anything listed as 'for:' is build parameters
+                    build_params = arg[4:]
+                    # EESSIBotBuildParams is essentially a dict, but parses the input argument
+                    # according to the expected argument format for 'for:'
+                    self.build_params = EESSIBotBuildParams(build_params)
+                else:
+                    # Anything that is not 'on:' or 'for:' should just be passed on as normal
+                    # No further parsing of the value is needed
+                    other_filter_args.extend([arg])
+
+            # If no 'on:' is found in the argument list, everything that follows the 'for:' argument
+            # (until the next space) is considered the argument list for the action filters
+            # Essentially, this represents a native build, i.e. the hardware we build on should be the
+            # hardware we build on
+            if not on_found:
+                for arg in cmd_as_list[1:]:
+                    if arg.startswith('for:'):
+                        # Extract everything after the 'for:' suffix and split by comma
+                        filter_content = arg[4:]
+                        target_args.extend(filter_content.split(','))
+
+            # Join the filter arguments and pass to EESSIBotActionFilter
+            # At this point, target_args is e.g. ["arch=amd/zen2","accel=nvidia/cc90"]
+            # But EESSIBotActionFilter expects e.g. "arch:amd/zen2 accel:nvidia/cc90"
+            # First, normalize to the ["arch:amd/zen2", "accel:nvidia/cc90"] format
+            normalized_filters = []
+            if target_args:
+                for filter_item in target_args:
+                    if '=' in filter_item:
+                        component, pattern = filter_item.split('=', 1)
+                        normalized_filters.append(f"{component}:{pattern}")
+
+            # Add the other filter args to the normalized filters. The other_filter_args are already colon-separated
+            # so no special parsing needed there
+            log(f"Extracted filter arguments related to hardware target: {normalized_filters}")
+            log(f"Other extracted filter arguments: {other_filter_args}")
+            normalized_filters += other_filter_args
+
+            # Finally, change into a space-separated string, as expected by EESSIBotActionFilter
+            # e.g "arch:amd/zen2 accel:nvidia/cc90 repo:my.repo.io"
+            if normalized_filters:
```

Comment:
Not sure if this can be easily changed, but here I again got the feeling that it would be nice to have all this build information (with `on` and `for`, etc) in some combined object, which could do the parsing and store it in such a way that you can easily get the right information out of it. It now feels like we are doing a lot of juggling around with those arguments in different places of the code. Perhaps it's not realistic and I'm overlooking things, but just wanted to throw out the idea.